### ISSUE TITLE
Rule proposal: comments on propTypes

### ISSUE THREAD

**Issue by alexzherdev** [yannickcr/eslint-plugin-react#1900]
When using tools like [react-docgen](https://github.com/reactjs/react-docgen) (or those based on it like [@storybook/addon-info](https://github.com/storybooks/storybook/tree/master/addons/info)) it is expected that each propType declaration has a comment with a description e.g.
```js
MyComponent.propTypes = {
  /**
   * Description of prop "foo".
   */
  foo: PropTypes.number
}
```
If the comment is not there, then the generated documentation is not complete/meaningful.

Would a rule that reminds developers of this be useful to have in the plugin, or is this too specific/special-case?

**Comment by ljharb** [comment]
I'm very much opposed to this for a few reasons: 
 - comments should not describe "what", only history/motivation; code should be self-documenting
 - the only commenting standard that has any traction is jsdoc, and this one seems like something unique and niche

**Comment by alexzherdev** [comment]
Generally speaking both of these make sense.

For your first point, FWIW, the use case I'm thinking of is a reusable component library that has a "showcase" website with the examples for those components (I'm sure you've seen those in the wild). So at this point you're not yet looking at / familiar with the source code per se, you just get a "visual" overview of what the components are and what their API looks like.
In the context of an application though, yeah, I don't usually feel the need to add comments like that for this very reason: components/props usually turn out self-documenting for when you're reading the code.

For the second point, I guess it's supposed to be compatible with the syntax of jsdoc, but probably not the generator part of it 😕 

Anyway, just wanted to check here first as I may need this for internal use. I saw that react-docgen is under the reactjs github org so I thought maybe this could be helpful to others. Feel free to close this then, sorry for the churn 👍

**Comment by ljharb** [comment]
What we usually do at airbnb is use https://www.npmjs.com/package/annotated-prop-types in a storybook-like environment.

**Comment by interactivellama** [comment]
If anyone is searching for this feature in the future, what we did in https://github.com/salesforce/design-system-react was use https://github.com/reactjs/react-docgen and scan the created JSON file for empty `description` strings.

See https://github.com/salesforce/design-system-react/blob/master/tests/component-docs-test.js
